tests/functional/splinter_test.c

```diff
@@ -2852,8 +2852,10 @@ splinter_test(int argc, char *argv[])
          platform_assert(SUCCESS(rc));
          break;
       case parallel_perf:
-         platform_assert(max_async_inflight == 0
-                         || task_system_use_bg_threads(ts));
+         platform_assert(
+            max_async_inflight == 0
+            || (0 < task_cfg.num_background_threads[TASK_TYPE_MEMTABLE]
+                && 0 < task_cfg.num_background_threads[TASK_TYPE_NORMAL]));
```

Comment:
I am a bit confused with this check.

I am not quite following what the new rules is. Previous to this PR, we needed both bg-thread config params to be 0 or both to be non-zero.

And I sort of understood that with this newest round you are relaxing that check. Aren't you now allowing only one  of the 2 params to be set?

If so, should the check on L2858 be `||` ?

Reply:
I actually don't know what is required by the async code, so I just conservatively required both to be non-zero, which is what was being required before anyway.